common/optimiser/Optimiser.java

```diff
@@ -56,12 +57,16 @@ public Optimiser() {
     }
 
     public synchronized Status optimise(long timeLimitMillis) {
+        if (hasSolver && constraintsChanged()) {
+            constraints.forEach(constraint -> constraint.initialise(solver));
+            status = Status.NOT_SOLVED;
+        }
         if (isOptimal()) return status;
         else if (!hasSolver) initialiseSolver();
         solver.setTimeLimit(timeLimitMillis);
         status = Status.of(solver.solve(parameters));
-        variables.forEach(OptimiserVariable::recordValue);
-        clearInitialisation();
+        if (status != Status.NOT_SOLVED && status != Status.ERROR)
```

Comment:
so it's our style to use only one of these two styles:
```
if (condition) single-line-operation;
```
or

```
if (condition) {
  cant fit in the same line as the condition operation
}
```
for loops we always use `{}`

I noticed this in a couple places in this PR that we can bring to consistency with the rest of the code base